packages/flutter_tools/test/general.shard/widget_preview/preview_code_generator_test.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/flutter_manifest.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/widget_preview/preview_code_generator.dart';
+import 'package:test/test.dart';
+
+import '../../src/context.dart';
+
+void main() {
+  group('$PreviewCodeGenerator', () {
+    late PreviewCodeGenerator codeGenerator;
+    late FlutterProject project;
+
+    setUp(() {
+      final FileSystem fs = MemoryFileSystem.test();
+      final FlutterManifest manifest = FlutterManifest.empty(logger: BufferLogger.test());
+      final Directory projectDir =
+          fs.currentDirectory.childDirectory('project')
+            ..createSync()
+            ..childDirectory('lib').createSync();
+      project = FlutterProject(projectDir, manifest, manifest);
+      codeGenerator = PreviewCodeGenerator(widgetPreviewScaffoldProject: project, fs: fs);
+    });
+
+    testUsingContext(
+      'correctly generates ${PreviewCodeGenerator.generatedPreviewFilePath}',
+      () async {
+        // Check that the generated preview file doesn't exist yet.
+        final File generatedPreviewFile = project.directory.childFile(
+          PreviewCodeGenerator.generatedPreviewFilePath,
+        );
+        expect(generatedPreviewFile.existsSync(), false);
+
+        // Populate the generated preview file.
+        codeGenerator.populatePreviewsInGeneratedPreviewScaffold(const <String, List<String>>{
+          'foo.dart': <String>['preview'],
+          'src/bar.dart': <String>['barPreview1', 'barPreview2'],
+        });
+        expect(generatedPreviewFile.existsSync(), true);
```

Comment:
Nit: package:file_testing provides an `exists` matcher that I assume would give us a better failure message here.